libtransmission/torrent.cc

```diff
@@ -1903,6 +1903,29 @@ void tr_torrentSetLabels(tr_torrent* tor, tr_labels_t&& labels)
 ****
 ***/
 
+void tr_torrentSetGroup(tr_torrent* tor, std::string_view group)
```

Comment:
Unless the code needs to be this way, I'd prefer this be a member function e.g.

```c++
void tr_torrent::setGroup(std::string_view group);

std::string const tr_torrent::group() const;
```

The existing C-style functions that you're taking inspiration from is just code that hasn't been refactored into member functions now that libtransmission is C++